src/app/codegen-interaction-model/CodegenDataModel.cpp

```diff
@@ -0,0 +1,530 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app/codegen-interaction-model/CodegenDataModel.h>
+
+#include <app-common/zap-generated/attribute-type.h>
+#include <app/RequiredPrivilege.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/endpoint-config-api.h>
+#include <lib/core/DataModelTypes.h>
+#include <optional>
+
+namespace chip {
+namespace app {
+namespace {
+
+/// Checks if the specified ember cluster mask corresponds to a valid
+/// server cluster.
+bool IsServerMask(EmberAfClusterMask mask)
```

Comment:
OK, so: please do a followup to have this be a method on EmberAfCluster.

Reply:
Updated to make this a member method.